python/tvm/relay/to_onnx.py

```diff
@@ -0,0 +1,422 @@
+        else:
+            output_name = node['input_names'][1]
+
+        inputs = [node['input_names'][0], output_name]
+        matmul_node = onnx.helper.make_node('Add', inputs, node['output_names'])
+        model_container.add_nodes([matmul_node])
+
+
+relay_to_onnx_op_mapping = {
+    'reshape': Reshape,
+    'conv2d': Conv,
+    'add': rename('Add'),
+    'relu': rename('Relu'),
+    'transpose': Transpose,
+    'dense': MatMul,
+    'max_pool2d': MaxPool,
+    'batch_flatten': Flatten,
+    'multiply': rename('Mul'),
+    'bias_add': BiasAdd,
+    'batch_norm': BatchNormalization,
+    'global_avg_pool2d': rename('GlobalAveragePool'),
+    'concatenate': Concat,
+    'dropout': Dropout,
+    'avg_pool2d': AveragePool,
+    'divide': rename('Div')
+}
+
+
+class ModelContainer(object):
+    """ A container class to hold  different attributes of ONNX model graph
+    """
+
+    def __init__(self, name):
+        self._name = name
+        self._inputs = []
+        self._outputs = []
+        self._nodes = []
+        self._initializers = []
+
+    def add_inputs(self, inputs):
+        self._inputs.extend(inputs)
+
+    def add_outputs(self, outputs):
+        self._outputs.extend(outputs)
+
+    def add_nodes(self, nodes):
+        self._nodes.extend(nodes)
+
+    def add_initializers(self, initializers):
+        self._initializers.extend(initializers)
+
+    def make_model(self):
+        graph_def = onnx.helper.make_graph(
+            self._nodes,
+            self._name,
+            self._inputs,
+            self._outputs,
+            self._initializers
+        )
+
+        return onnx.helper.make_model(graph_def, producer_name='relay')
+
+
+class RelayToONNXConverter(object):
+    """A helper class converting topologically sorted  Relay Node list to ONNX model
+
+    Parameters
+    ----------
+    name : str
+       name of the model
+
+    node_list : list
+        topologically sorted Relay Node entry list
+    """
+
+    def __init__(self, name, node_list, params):
+        self._name = {}
+        self._mc = ModelContainer(name)
+        self._node_list = node_list
+        self._params = params
+
+    def convert_to_onnx(self):
+        """ Loop through topologically sorted list of Relay nodes and generate a ONNX model"""
+        for idx, node_entry in enumerate(self._node_list):
+            out_idx = idx
+            node = node_entry['node']
+            if isinstance(node, Call):
+                self._add_node(node_entry, idx)
+            elif isinstance(node, Var):
+                self._add_input(node_entry, idx)
+            elif isinstance(node, Constant):
+                self._add_constant_input(node_entry, idx)
+            elif isinstance(node, (TupleGetItem, Tuple)):
+                out_idx = idx - 1  # Need to work on this. No equivalent ONNX operator found yet
+            else:
+                raise NotImplementedError("Relay Node of type {0} is not "
```

Comment:
node could be other types as well, i.e. Tuple/TupleWrapper, etc.

Reply:
Currently TuppleWrapper is not supported.  Tuple is supported.